changelog.d/20220418_231901_sirosen_use_click.rst

```diff
@@ -0,0 +1,11 @@
+Changed
+^^^^^^^
+
+- The CLI interface for `funcx-endpoint` has been updated in several ways:
+
+  - The optional name argument is now an option, `-n/--name`.
+    Rewrite `funcx-endpoint start NAME` to `funcx-endpoint start -n NAME`
```

Comment:
Could this change be avoided? `funcx-endpoint start/stop <ENDPOINT_NAME>` is the most frequently used cmdline invocation I do.

Reply:
It can be avoided, but I think it's basically incorrect design-wise.
Optional things should be options.

Reply:
Since you're raising this in multiple places, and I'm less strongly attached to it than to the other changes in interface, I'm happy to revert this part. We can have an optional positional argument. I'll go on the record, however, as saying that I believe that optional positionals often make CLIs harder to use, and I'm generally against the pattern.

Reply:
Revert done and just pushed. Marking resolved.